src/sentry/api/endpoints/user_notification_details.py

```diff
@@ -24,20 +56,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
         return results
 
     def serialize(self, obj, attrs, user, *args, **kwargs):
-        data = {option.key: option.value for option in attrs}
 
-        return {
-            'alertEmail': data.get('alert_email'),
-            'deployEmails': data.get('deploy-emails'),
-            'seenReleaseBroadcast': data.get('seen_release_broadcast'),
-            'selfAssignIssue': data.get('self_assign_issue'),
-            'selfNotifications': data.get('self_notifications'),
-            'subscribeByDefault': data.get('subscribe_by_default'),
-            'workflowNotifications': data.get('workflow:notifications'),
-        }
+        raw_data = {option.key: option.value for option in attrs}
+
+        data = {}
+        for key in USER_OPTION_SETTINGS:
+            uo = USER_OPTION_SETTINGS[key]
+            val = raw_data.get(uo['key'], uo['default'])
+            if (uo['type'] == bool):
+                data[key] = bool(int(val))  # '1' is true, '0' is false
+            elif (uo['type'] == int):
+                data[key] = int(val)
+
+        return data
+
+
+class UserNotificationDetailsSerializer(serializers.Serializer):
+    deployNotifications = serializers.IntegerField(required=False, min_value=2, max_value=4)
+    personalActivityNotifications = serializers.BooleanField(required=False)
+    selfAssignOnResolve = serializers.BooleanField(required=False)
+    subscribeByDefault = serializers.BooleanField(required=False)
+    workflowNotifications = serializers.IntegerField(required=False, min_value=0, max_value=2)
 
 
 class UserNotificationDetailsEndpoint(UserEndpoint):
     def get(self, request, user):
         serialized = serialize(user, request.user, UserNotificationsSerializer())
         return Response(serialized)
+
+    def put(self, request, user):
+        serializer = UserNotificationDetailsSerializer(data=request.DATA)
+
+        if serializer.is_valid():
+            for key in serializer.object:
+                db_key = USER_OPTION_SETTINGS[key]['key']
+                val = six.text_type(int(serializer.object[key]))
+                (uo, created) = UserOption.objects.get_or_create(user=user, key=db_key, project=None)
+                uo.update(value=val)
```

Comment:
not a good idea for this code, but i realized per our discussion earlier that you could achieve some degree of bulk updates by batching them by value.